### DISCUSSION TITLE
How would you configure this library in a DI driven application?

### DISCUSSION THREAD

**Discussion by silvanknecht** [thepirat000/Audit.NET#689] [Q&A]
Our current applications are all DI-driven and I'd like to know what you would assume is the cleanest way to configure this static service in a DI-driven world (with HostBuilder, ConfigureService(), etc...). The main reason is that we want to use the the option pattern to configure the service.

**Comment by thepirat000** [comment]
It depends on which extensions are you using to generate the events. 

Some of the extensions for generating events, such as [Audit.EntityFramework.Core](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/README.md), [Audit.WebApi.Core](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.WebApi/README.md), [Audit.Mvc.Core](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.Mvc/README.md), and [Audit.SignalR](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.SignalR/README.md), will automatically attempt to resolve the `IAuditScopeFactory` and `AuditDataProvider` instances from the service provider, so you should be able to register your scope factory and data provider in your start up, for example:

```c#
services.AddScoped<IAuditScopeFactory, AuditScopeFactory>();
services.AddSingleton<AuditDataProvider>(new FileDataProvider(cfg => cfg.Directory(@"C:\Logs"))
```


https://github.com/thepirat000/Audit.NET/blob/master/README.md#audit-scope-factory
